testing/internal/e2e/vault/vault.go

```diff
@@ -31,71 +34,108 @@ func loadConfig() (*config, error) {
 
 // Setup verifies if appropriate credentials are set and adds the boundary controller
 // policy to vault. Returns the vault address.
-func Setup(t testing.TB) (string, string) {
+func Setup(t testing.TB) (vaultAddr string, boundaryPolicyName string, kvPolicyFilePath string) {
 	c, err := loadConfig()
 	require.NoError(t, err)
+	vaultAddr = c.VaultAddr
 
-	_, filename, _, ok := runtime.Caller(0)
-	require.True(t, ok)
-	ctx := context.Background()
-	policyName := "boundary-controller"
-	output := e2e.RunCommand(ctx, "vault",
-		e2e.WithArgs(
-			"policy", "write", policyName,
-			path.Join(path.Dir(filename), "boundary-controller-policy.hcl"),
-		),
-	)
-	require.NoError(t, output.Err, string(output.Stderr))
-	t.Cleanup(func() {
-		output := e2e.RunCommand(ctx, "vault",
-			e2e.WithArgs("policy", "delete", policyName),
-		)
-		require.NoError(t, output.Err, string(output.Stderr))
-	})
+	// Set up boundary policy
+	data, err := boundaryPolicyFile.ReadFile("testdata/boundary-controller-policy.hcl")
+	require.NoError(t, err)
+	boundaryPolicyFilePath := fmt.Sprintf("%s/%s", t.TempDir(), "boundary-controller-policy.hcl")
+	f, err := os.Create(boundaryPolicyFilePath)
+	require.NoError(t, err)
+	_, err = f.WriteString(string(data))
+	require.NoError(t, err)
```

Comment:
Looks like we can avoid this whole write-out-to-tmp-file-dance by just sending the policy contents to stdin: https://developer.hashicorp.com/vault/docs/commands/policy/write. Would be nice to avoid all the file system operations that can fail. What do you think?

Reply:
Can give it a shot

Reply:
Do you know if we can do the same for boundary/postgres? My initial search didn't turn up with anything, which means we would have to do this dance in other places.

Reply:
@johanbrandhorst After thinking about it some more, I think I'd like to leave it as is since this aligns with the what our tutorials do (creating a file).

Reply:
Hm, in that case, maybe we should just not bother embedding and read straight from the testdata directory?

Reply:
I don't mind keeping the embedding in there as it helped with the issue of needing to use `runtime.Caller` to get the path of this file since you could have different callers here. Is there another solution?

Reply:
Hm, I guess maybe the Vault command doesn't know to look in the testdata folder... alright sure!